src/entities/protocols/uniswap.ts

```diff
@@ -63,7 +67,7 @@ export class UniswapTrade implements Command {
         this.trade.maximumAmountIn(this.options.slippageTolerance).quotient.toString(),
       ])
       // since WETH is now owned by the router, the router pays for inputs
-      payerIsUser = false
+      Object.assign(this.options, { payerIsUser: false })
```

Comment:
editing a param (this.options) is a bit of an antipattern as it can cause unexpected behavior for callers (i.e. the object they sent in changes in-place). Consider just having a `payerIsUser` field that is set to `true || this.options.payerIsUser`

Reply:
Even better potentially, if this is the only case where we have to modify payerIsUser, we could definitely set it in the constructor and make it a readonly param to capture all potential cases in one place. Also maybe extra specification that options.payerIsUser is specifically because tokens were directly sent to de-conflate the ideas i.e.

```
constructor().. {
   if (this.trade.requiresWrap()) this.payerIsUser = false;
   if (!this.options.directSend) this.payerIsUser = false;
   else this.payerIsUser = true;
}
```
or something

Reply:
Ah ok yeah I see what you mean - pushed up something similar in latest commit